src/main/java/com/infobip/jira/JiraVersionGenerator.java

```diff
@@ -1,130 +1,108 @@
-                                @Nonnull Commit releaseChangeset,
-                                @Nonnull Iterator<Commit> changesetIterator,
-                                @Nonnull CommitMessageVersionExtractor commitMessageVersionExtractor) {
+    public JiraVersionGenerator(JiraService jiraService,
+                                Commit releaseChangeset,
+                                Iterator<Commit> changesetIterator,
+                                CommitMessageVersionExtractor commitMessageVersionExtractor,
+                                Clock clock) {
 
         this.releaseChangeset = releaseChangeset;
 
         this.jiraService = requireNonNull(jiraService);
         this.changesetIterator = requireNonNull(changesetIterator);
         this.commitMessageVersionExtractor = requireNonNull(commitMessageVersionExtractor);
+        this.clock = clock;
     }
 
-    public void generateJiraVersionAndLinkIssues(@Nonnull String jiraVersionPrefix, @Nonnull ProjectKey projectKey) throws IOException, CredentialsRequiredException, ResponseException {
+    public void generateJiraVersionAndLinkIssues(String jiraVersionPrefix,
+                                                 ProjectKey projectKey) {
 
         if (!changesetIterator.hasNext()) {
-            logger.warn("no changesets in commit");
             return;
         }
 
-        Optional<String> versionName = commitMessageVersionExtractor.extractVersionName(releaseChangeset.getMessage());
-
-        if (!versionName.isPresent()) {
-            return;
-        }
-
-        List<Commit> versionChangesets = getVersionChangesets();
-
-        Version version = Version.of(jiraVersionPrefix + versionName.get(),
-                                     projectKey,
-                                     true,
-                                     releaseChangeset.getAuthorTimestamp());
-
-        try {
-            if (jiraService.doesJiraVersionExist(version)) {
-                logger.info(String.format("jira version %s already exists", version));
-                return;
-            }
-        } catch (CredentialsRequiredException | IOException | ResponseException e) {
-            logger.error(String.format("failed to verify if Jira version %s already exists", version), e);
-            throw e;
-        }
+        commitMessageVersionExtractor.extractVersionName(releaseChangeset.getMessage())
+                .ifPresent(versionName -> generateJiraVersionAndLinkIssues(jiraVersionPrefix, projectKey, versionName));
 
-        try {
-            jiraService.createJiraVersion(version);
-        } catch (CredentialsRequiredException | IOException | ResponseException e) {
-            logger.error("failed to create Jira version " + version, e);
-            throw e;
-        }
+    }
 
-        List<IssueKey> issueKeys = getIssueKeys(versionChangesets, projectKey);
+    private void generateJiraVersionAndLinkIssues(String jiraVersionPrefix,
```

Comment:
can just be `generate` - better than a long name which is still missing a part (version releasing)
same for the public method above (the class name already offers context)